project/src/com/google/daggerquery/executor/models/Query.java

```diff
@@ -99,26 +107,25 @@ public Query(String typeName, String... parameters) {
    * may look like this: "com.google.First -> com.google.Second -> com.google.Third".
    * </ul>
    *
+   * @throws MisspelledNodeNameException if the specified source node contains a typo
+   * and can be corrected in no more than three steps
+   *
    * @throws IllegalArgumentException if specified source node doesn't exist
    */
   public List<String> execute(BindingGraph bindingGraph) {
     switch (name) {
       case DEPS_QUERY_NAME: {
         String source = parameters[0];
 
-        if (!bindingGraph.getAdjacencyListMap().containsKey(source)) {
-          throw new IllegalArgumentException(String.format("Specified source node %s doesn't exist.", source));
-        }
+        checkNodeForCorrectness(source, bindingGraph);
```

Comment:
Since it looks like we're checking in every case of the switch statement with the same arguments, can we move this to be above the switch statement?

Reply:
This will force us to take the first source parameter and pass it to the function, but in future queries, the source node may not be passed as the first parameter. 

We [discussed this here in a previous PR](https://github.com/googleinterns/dagger-query/pull/15/files/27d926ba84abfe8e00f5bfe9d6ca19c4acbf677d#r470689088) and decided it might be better to repeat some logic.

Reply:
Thanks for linking that conversation, I defer to Karl's expertise :)